samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -1128,6 +1145,7 @@ class SamzaContainer(
     }
 
     taskInstances.values.foreach(_.shutdownTask)
+    taskInstances.values.foreach(_.shutdownContext)
```

Comment:
Might be nicer if the relationship b/w init+close and context start+stop is symmetric. I.e., Either the context is started before and stopped after the task init and close respectively, or vice versa.

Reply:
Good catch.